build: add teamcity-bless-provisional-artifacts.sh to replace TeamCity script

otan opened the pull request:
Copy the TC script that runs bless into a shell file so that bless can
now be code reviewed by other members.

Intending to backport this to 20.1, 19.2 and 19.1.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50901)
<!-- Reviewable:end -->

jlinder reviewed (approved):
One suggestion: change the name of the script to `build/teamcity-bless-provisional-artifacts.sh` to match the name of the build configuration. 

LGTM

otan committed:
build: add teamcity-bless-provisional-artifacts.sh to replace TeamCity script

Copy the TC script that runs bless into a shell file so that bless can
now be code reviewed by other members.

Intending to backport this to 20.1, 19.2 and 19.1.

Release note: None

otan commented:
TFTR!

bors r=jlinder
